test/testlib.py

```diff
@@ -970,44 +968,32 @@ def test_main(argv=None, suite=None, attachments=None):
     sys.stdout.flush()
     sys.stdout = os.fdopen(sys.stdout.fileno(), 'w', 0)
 
-    parser = argparse.ArgumentParser(description='Run Cockpit test(s)')
-    parser.add_argument('-j', '--jobs', dest="jobs", type=int,
-                        default=os.environ.get("TEST_JOBS", 1), help="Number of concurrent jobs")
-    parser.add_argument('-v', '--verbose', dest="verbosity", action='store_const',
-                        const=2, help='Verbose output')
-    parser.add_argument('-t', dest='trace', action='store_true',
-                        help='Trace machine boot and commands')
-    parser.add_argument('-q', '--quiet', dest='verbosity', action='store_const',
-                        const=0, help='Quiet output')
-    parser.add_argument('--thorough', dest='thorough', action='store',
-                        help='Thorough mode, no skipping known issues')
-    parser.add_argument('-s', dest='sit', action='store_true')
-    parser.add_argument('tests', nargs='*')
-
-    parser.set_defaults(verbosity=1)
-    args = parser.parse_args(argv)
-
-    if args.sit and args.jobs > 1:
+    standalone = opts is None
+    if standalone:
+        parser = testinfra.arg_parser()
```

Comment:
Now `parser` is only defined when running standalone.  `check-verify foo` will thus crash when trying to raise "tests may not be specified when ...".